### ISSUE TITLE
[Research Task] Investigate ways to include CNF, BFDB, OFDB, NCCDB, etc

### ISSUE THREAD

**Issue by gamesguru** [gg-archive/ntdb-old#7]
Currently, for v2, we dropped support down to USDA `SR` (standard reference) database to simplify the user experience and not make _our_ work too difficult up front.

Eventually we may want to **add support for more sources of data**.  This presents challenges in several ways
- Initially in getting it into an `SQL` image,
- Performance (e.g. searching 1.4 million food names),
- Customer experience (I searched blueberries, but I can't even find one from USDA because of so much "clutter" from OFDB).  **GRAPH THEORY TIME**

There is some **work _being_ done to include special interests** (flavonoid `flav/isoflav/proanth` database) [screenshot 2] but as for now, the other sources [screenshot 1] are _not_ being worked on or actively researched.

BFDB, CNF, OFDB
![Screenshot from 2020-01-24 14-08-32](https://user-images.githubusercontent.com/30691680/73207321-88e2c080-4112-11ea-8ab3-5b7eb659944d.png)

Flavonoids, Isoflavonoids, Proanthocyanins
![Screenshot from 2020-01-27 14-33-30](https://user-images.githubusercontent.com/30691680/73207333-8bddb100-4112-11ea-9d9f-10c8d23f06ca.png)

Additional info from `beta` in `data/special-interests.rel.txt`:

```
[golden]
Nutr_no,rda,Units,tagname,NutrDesc

[intermediate]
id,units,name


[isoflav]
"Nutr_no","NutrDesc","Unit"

[proanth]
"Nutr_No","Units","Tagname","NutrDesc","Num_Dec","SR_Order"

[flav]
"Nutr_no","Nutrient name","Flav_Class","Tagname","Unit"



curerntly down for maintanence:
https://data.nal.usda.gov/dataset/usda-special-interest-databases-flavonoids
```

**Comment by jorbecalona** [comment]
They need attribute mappings, just like LDAP does. Then you can relate them as such, and huzzah.

**Comment by gamesguru** [comment]
yeah they need to be hardcoded somewhere.  No consistency either.  `proanth` has "Units" the others have "Unit"

the bigger issue i guess is upgrading the SR database to a newer revision (when it comes out), and still preserve all the user data potentially built off it

this one is also interesting
https://github.com/gamesguru/ntdb/issues/14
